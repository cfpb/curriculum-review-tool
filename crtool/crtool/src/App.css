.App {
  text-align: center;
}

.header-footer-format {
  text-align: center;
}

html,body{
  height: 100%
}

#holder{
  min-height: 100%;
  position:relative;
}

#footer{
  height: 100px; 
  width:100%;
  position: absolute;
  left: 0;
  bottom: 0; 
}

.DistinctiveBoxFormat {
  text-align: left;
}

.CriterionPageTextFormat {
  text-align: left;
}

.Distinctive {
	float: left;
	width: 250px;
	height: 140px;
	margin: 0.5em;
	border: 1px solid black;
	padding: 0.5em;
}

.DistinctivesBlock:after {
  content: "";
  display: block;
  clear: both;
}

.SurveyPage {
  margin-top: 30px;
}

.a-text-input {
  width: 400px;
}