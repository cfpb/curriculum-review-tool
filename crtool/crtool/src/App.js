import 'react-app-polyfill/ie9'; // For IE 9-11 support
import 'react-app-polyfill/stable';
import 'core-js/es6/map';
import 'core-js/es6/set';
import React, { Component } from 'react';

import CustomerReviewToolComponent from "./js/components/CustomerReviewToolComponent";

class App extends Component {
  render() {
    return (
      <div className="App">
        <CustomerReviewToolComponent />
      </div>
    );
  }
}

export default App;
