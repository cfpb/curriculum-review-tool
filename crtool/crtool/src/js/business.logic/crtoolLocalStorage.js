const crtoolLocalStorage = {

    getCurrentReview() {
        let review = false;
        let review_id = this.getUrlParameter('token') || localStorage.getItem('curriculumReviewId') || '';
        console.log('getCurrentReview: line 6');
        console.log(review_id);
//        if (!review_id) {
//            review = this.getReviewFromLocalStorage();
//            review_id = "id" in review ? review.id : null;
//            console.log(review_id);
//        }
//        else {
//            review = this.getReviewFromLocalStorage(review_id);
//        }
//        console.log(review)
        review = this.loadReviewFromDatabase(review_id);
        console.log(review);
        return review;
    },

    /*
     * load review state from LocalStorage
     */
    getReviewFromLocalStorage(review_id = null) {
        let review = {};
        console.log('getReviewFromLocalStorage review: line 27');
        if (!review_id) {
            review_id = localStorage.getItem('curriculumReviewId') || "";
            console.log('review: line 30');
            console.log(review_id);
        }
        if (review_id) {
            review = localStorage.getItem('crtool.' + review_id) || "{}";
            console.log('review: line 35');
            if (review) {
                localStorage.setItem('curriculumReviewId', review_id)
                review = JSON.parse(review);
                console.log('review: line 40');
                console.log(review);
            }
        }
        return review;
    },

    /*
     * load review state from Database
     */
    loadReviewFromDatabase(review_id) {
        console.log('loadReviewFromDatabase: line 52');
        let review = this.getReviewFromLocalStorage(review_id);
        const xhttp = new XMLHttpRequest();
        xhttp.open("GET", "../get-review?tdp-crt_id=" + review_id, false);
        xhttp.send();
        if (xhttp.readyState === 4 && xhttp.status === 200) {
            const db_review = JSON.parse(xhttp.responseText);
            console.log('db_review: line 59');
            console.log(db_review);
            console.log('review: line 61');
            console.log(review);
            const is_review_valid = ("last_updated" in review) && ("id" in review)
            const is_db_review_valid = ("last_updated" in db_review) && ("id" in db_review)
            if (is_review_valid && is_db_review_valid) {
                console.log('db_review.last_updated: line 66');
                console.log(db_review.last_updated);
                console.log(new Date(db_review.last_updated));
                console.log('review.last_updated: line 68');
                console.log(review.last_updated);
                console.log(new Date(review.last_updated));
            }
            if (!is_review_valid ||
                (
                    is_review_valid &&
                    is_db_review_valid &&
                    new Date(db_review.last_updated) > new Date(review.last_updated)
                )
            ) {
                console.log('crtoolLocalStorage.js: line 78');
                review = db_review;
                localStorage.setItem('crtool.' + db_review.id, JSON.stringify(db_review));
                localStorage.setItem('curriculumReviewId', db_review.id)
            }
        }
        return review;
    },

    /*
     * Overwrite Database review state with state in LocalStorage
     */
    updateDatabaseReview(review) {
        console.log('updateDatabaseReview: line 91');
        const xhttp = new XMLHttpRequest();
        xhttp.open("POST", "../update-review/", false);
        xhttp.setRequestHeader("Content-Type", "application/json;charset=UTF-8");
        console.log('review: line 95');
        console.log(review);
        xhttp.send(JSON.stringify(review));
        if (xhttp.readyState === 4 && xhttp.status === 200) {
            review = JSON.parse(xhttp.responseText);
            console.log('review: line 100')
            console.log(review);
            if ("id" in review) {
                console.log('update localStorage: line 103');
                localStorage.setItem('crtool.' + review.id, JSON.stringify(review));
            }
        }
        return review;
    },

    setItem(key, value) {
        console.log('crtoolLocalStorage setItem: line 111');
        console.log(key + ': ' + value);
        const review = this.getCurrentReview();
        if (review) {
            console.log('setItem review: line 115');
            review[key] = value;
            return this.updateDatabaseReview(review);
        }
        return false;
    },

    getItem(key) {
        console.log('crtoolLocalStorage getItem: line 123');
        console.log(key);
        const review = this.getCurrentReview();
        if (key in review) {
            console.log('getItem: line 127');
            return review[key];
        }
        return null;
    },

    removeItem(key) {
        const review = this.getCurrentReview();
        if (key in review) {
            delete review[key];
            return true;
        }
        return false;
    },

    clear() {
        const review = this.getCurrentReview();
        if (review) {
            localStorage.removeItem('curriculumReviewId');
        }
    },

    getUrlParameter(name) {
        name = name.replace(/[\[]/, '\\[').replace(/[\]]/, '\\]');
        const regex = new RegExp('[\\?&]' + name + '=([^&#]*)');
        const results = regex.exec(window.location.search);
        return results === null ? '' : decodeURIComponent(results[1].replace(/\+/g, ' '));
    },

};

export default crtoolLocalStorage;