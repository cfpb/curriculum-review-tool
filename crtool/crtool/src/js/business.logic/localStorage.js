const localStorage = {

    setItem(uuid, key, value) {
        xhttp.open("POST", "demo_post.asp", true);
        xhttp.send();
    },

    getItem(uuid, key) {
        xhttp.open("GET", "demo_post.asp", true);
        xhttp.send();
    },

    removeItem(uuid, key) {

    },

    clear(uuid) {

    }

};

export default localStorage;