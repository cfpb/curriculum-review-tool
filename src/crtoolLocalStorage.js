import C from "./js/business.logic/constants";

const CHECK_FREQUENCY = 60e3;

/**
 * A proxy for the localStorage API which periodically saves to the DB and
 * can synchronize itself when the page loads.
 */
const ls = {
    // Use DOM LocalStorage, but allow swapping
    localStorage: localStorage,

    review: {},
    lastTimeSaved: '',
    timeoutHandle: 0,
    testing: false,

    redirectHome() {
        if (ls.testing) {
            return;
        }

        window.location.href = C.START_PAGE_RELATIVE_URL;
    },

    /**
     * Set up for testing, injecting mock localStorage
     *
     * @param {WindowLocalStorage} localStorage
     */
    initForTesting(localStorage) {
        ls.localStorage = localStorage;
        ls.testing = true;
        ls.review = {
            id: 'testing',
        };
    },

    async init() {
        const token = ls.getUrlParameter('token');
        let review_id = token || ls.localStorage.getItem('curriculumReviewId') || '';
        if (!review_id) {
            return ls.redirectHome();
        }

        // If review_id found but a token is not set, update the URL
        if (!token && review_id) {
            const new_url = ls.updateUrlParameter(window.location.href, 'token', review_id);
            window.history.pushState({ path: new_url }, '', new_url);
        }

        const ls_review = ls.loadReviewFromLocalStorage(review_id);

        let db_review;
        try {
            db_review = await ls.fetchReviewFromServer(review_id);
            if (!db_review) {
                // Do not allow end users to set their own ID via the URL or localStorage.
                ls.localStorage.removeItem('crtool.' + review_id);
                ls.localStorage.removeItem('curriculumReviewId');
                ls.redirectHome();
                return;
            }
            if (!db_review.ls_modified_time) {
                // A fresh review. Let's set a local time on it
                db_review.ls_modified_time = (new Date()).toISOString();
            }
        } catch (e) {
            console.error(e);
            // Request failed, just try to carry on with local storage
        }

        if (!ls_review && !db_review) {
            ls.redirectHome();
            return;
        }

        if (ls_review && ls_review.ls_modified_time > db_review.ls_modified_time) {
            // Trust that newer local time means more up-to-date data
            ls.review = ls_review;
            ls.localStorage.setItem('curriculumReviewId', review_id);
            ls.timeoutHandle = setTimeout(ls.saveIfDirty, 0);
        } else {
            ls.review = db_review;
            // Just bring local storage up to speed
            ls.localStorage.setItem('crtool.' + db_review.id, JSON.stringify(db_review));
            ls.localStorage.setItem('curriculumReviewId', db_review.id);
            // Don't need to sync immediately after load
            ls.lastTimeSaved = ls.review.ls_modified_time;
            ls.timeoutHandle = setTimeout(ls.saveIfDirty, CHECK_FREQUENCY);
        }
    },

    isReady() {
        return Boolean(ls.review.id);
    },

    saveReviewToLocalStorage() {
        ls.review.ls_modified_time = (new Date()).toISOString();
        ls.localStorage.setItem('crtool.' + ls.review.id, JSON.stringify(ls.review));
    },

    isValidReview(review) {
        if (typeof review !== 'object' || review === null) {
            return false;
        }

        for (const prop of ['id', 'last_updated']) {
            if (!review.hasOwnProperty(prop)) {
                return false;
            }
        }

        return true;
    },

    /**
     * Returns a Promise with a valid review or undefined.
     *
     * Rejects if request fails. Invalid response resolves to undefined.
     */
    async fetchReviewFromServer(review_id) {
        if (ls.testing) {
            return;
        }

        const xhttp = new XMLHttpRequest();
        xhttp.open("GET", "../get-review?tdp-crt_id=" + review_id, true);
        xhttp.timeout = 5e3;

        return new Promise((resolve, reject) => {
            xhttp.onreadystatechange = () => {
                if (xhttp.readyState !== 4) {
                    return;
                }

                if (xhttp.status !== 200) {
                    console.error('Review fetch failed.');
                    reject();
                }

                try {
                    const review = JSON.parse(xhttp.responseText);
                    resolve(ls.isValidReview(review) ? review : undefined);
                } catch (e) {
                    console.error(e);
                    reject();
                }
            };

            xhttp.send();
        });
    },

    /*
    * Returns a valid review from LocalStorage or undefined
    */
    loadReviewFromLocalStorage(review_id) {
        let review = ls.localStorage.getItem('crtool.' + review_id) || "{}";
        try {
            review = JSON.parse(review);
            if (!ls.isValidReview(review)) {
                throw new Error('invalid review');
            }

            return review;
        } catch (e) {
            console.error(e);
        }
    },

    /*
     * Overwrite Database review state with state in LocalStorage
     *
     * Rejects anytime a valid response was not received
     */
    async saveReviewToServer() {
        if (ls.testing) {
            ls.localStorage.setItem('crtool.' + ls.review.id, JSON.stringify(ls.review));
            return;
        }

        const xhttp = new XMLHttpRequest();
        xhttp.open("POST", "../update-review/", true);
        xhttp.timeout = 5e3;
        xhttp.setRequestHeader("Content-Type", "application/json;charset=UTF-8");

        return new Promise((resolve, reject) => {
            xhttp.onreadystatechange = () => {
                if (xhttp.readyState !== 4) {
                    return;
                }

                if (xhttp.status !== 200) {
                    console.error('DB save failed.');
                    reject();
                    return;
                }

                const review = JSON.parse(xhttp.responseText);
                if (ls.isValidReview(review)) {
                    console.info('Received response', ls);

                    if (ls.review.ls_modified_time > ls.lastTimeSaved) {
                        console.info('Scheduling re-save...');
                        // Leave dirty and don't overwrite local storage
                    } else {
                        ls.review = review;
                        ls.localStorage.setItem('crtool.' + review.id, JSON.stringify(ls.review));
                    }

                    resolve();
                    return;
                }

                console.error('Invalid server response');
                reject();
            };

            ls.lastTimeSaved = ls.review.ls_modified_time;
            console.info(`Saving version ${ls.lastTimeSaved}`);
            xhttp.send(JSON.stringify(ls.review));
        });
    },

    saveIfDirty() {
        // If called from an imperative action, we don't want to clear the existing schedule
        if (ls.timeoutHandle) {
            clearTimeout(ls.timeoutHandle);
        }

        if (ls.review.ls_modified_time > ls.lastTimeSaved) {
            // console.log('auto saving to database now');
            ls.saveReviewToServer();
        }

        ls.timeoutHandle = setTimeout(ls.saveIfDirty, CHECK_FREQUENCY);
    },

    // IE compatible method for getting a querystring parameter from a URL
    // Credit: https://stackoverflow.com/a/901144
    getUrlParameter(name) {
        name = name.replace(/[\[]/, '\\[').replace(/[\]]/, '\\]');
        const regex = new RegExp('[\\?&]' + name + '=([^&#]*)');
        const results = regex.exec(window.location.search);
        return results === null ? '' : decodeURIComponent(results[1].replace(/\+/g, ' '));
    },

    // Add / Update a key-value pair in the URL query parameters
    // Credit: https://gist.github.com/niyazpk/f8ac616f181f6042d1e0
    updateUrlParameter(uri, key, value) {
        // remove the hash part before operating on the uri
        const i = uri.indexOf('#');
        const hash = i === -1 ? ''  : uri.substr(i);
        uri = i === -1 ? uri : uri.substr(0, i);

        const re = new RegExp("([?&])" + key + "=.*?(&|$)", "i");
        const separator = uri.indexOf('?') !== -1 ? "&" : "?";
        if (uri.match(re)) {
            uri = uri.replace(re, '$1' + key + "=" + value + '$2');
        } else {
            uri = uri + separator + key + "=" + value;
        }
        return uri + hash;  // finally append the hash as well
    },

    setItem(key, value) {
        if (!ls.isReady()) {
            throw new Error('Use before init');
        }

        // console.log('crtoolLocalStorage setItem: line 111');
        // console.log(key + ': ' + value);
        // Only setItem value if it's different than what's already there.
        if (ls.review[key] !== value) {
            ls.review[key] = value;
            ls.saveReviewToLocalStorage();
        }
    },

    getItem(key) {
        if (!ls.isReady()) {
            throw new Error('Use before init');
        }

        if (ls.review.hasOwnProperty(key)) {
            // console.log('getItem: HIT');
            return ls.review[key];
        } else {
            // console.log('getItem: MISS');
            return null;
        }
    },

    removeItem(key) {
        if (!ls.isReady()) {
            throw new Error('Use before init');
        }

        if (ls.review.hasOwnProperty(key)) {
            delete ls.review[key];
            ls.saveReviewToLocalStorage();
        }
    },

    clear() {
        if (!ls.isReady()) {
            throw new Error('Use before init');
        }

        ls.localStorage.removeItem('curriculumReviewId');
    },
};

const crtoolLocalStorage = ls;

export default crtoolLocalStorage;
