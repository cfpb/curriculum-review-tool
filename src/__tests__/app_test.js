import React from 'react';
import ReactDOM from 'react-dom';
import localStorage from 'mock-local-storage';
import renderer from 'react-test-renderer';
import jest from 'jest';

import crtoolLocalStorage from '../crtoolLocalStorage';
import App from '../App';

crtoolLocalStorage.initForTesting(localStorage);

it('renders without crashing', () => {
  const component = renderer.create(
    <App />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
