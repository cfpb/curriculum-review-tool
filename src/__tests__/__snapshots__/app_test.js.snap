// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <div
    className="l-survey-top"
  >
    <button
      className="a-btn a-btn__link a-btn__no-line"
      data-gtm_ignore="true"
      onClick={[Function]}
    >
      <span
        className="a-icon a-icon__large a-icon__space-after"
      >
        <svg
          className="cf-icon-svg"
          viewBox="0 0 1000 1200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M500 105.2c-276.1 0-500 223.9-500 500s223.9 500 500 500 500-223.9 500-500-223.9-500-500-500zm59.5 816.2c-6.5 15.2-18.6 27.3-33.7 33.7l-.3.1c-7.7 3.2-15.9 4.9-24.2 4.9-25.8.2-49.2-15.1-59.3-38.9-10.2-23.7-4.8-51.2 13.6-69.3 6-5.7 12.9-10.3 20.6-13.4 7.9-3.3 16.5-5 25.1-5 8.3 0 16.5 1.7 24.2 5l.2.1c7.5 3.2 14.3 7.7 20.2 13.4 18.3 18.2 23.6 45.6 13.6 69.4zm115.7-430.5c-4.1 15.2-10.1 29.9-17.7 43.7-7 12.4-15.1 24.2-24.2 35.1-8.8 10.4-17.3 20.3-25.4 29.5-.4.4-.8.8-1.2 1.3-7.6 8-14.6 15.6-20.8 22.9-6.6 7.6-12.4 15.9-17.4 24.7-5.2 9.2-9.1 19.1-11.8 29.3-2.8 10.7-4.3 23.2-4.3 37.2v2.2c0 27.4-22.4 49.8-50 49.8s-50-22.4-50-50v-2.2c0-22.7 2.6-43.8 7.6-62.9 4.9-18.6 12.1-36.4 21.6-53.2 8.2-14.3 17.7-27.8 28.5-40.3 7.1-8.3 15.1-17 23.6-26 7.3-8.3 15.1-17.3 23.1-26.8 5.2-6.3 9.8-13 13.8-20.1 3.6-6.7 6.4-13.8 8.4-21.2 1.4-5.1 3.1-14.1 3.1-28.4a80.7 80.7 0 0 0-6.2-32.5c-4.1-10.1-10.1-19.2-17.7-27-7.5-7.6-16.5-13.7-26.4-17.8-9.7-4.2-20.1-6.2-30.7-6-10.9-.2-21.7 1.9-31.7 6.2-20.5 8.5-36.7 24.8-45.2 45.3-4.3 10-6.4 20.9-6.2 31.8v1.2c0 27.6-22.4 50-50 50s-50-22.4-50-50v-1.3c0-25.5 4.7-48.8 14.3-71.2 18.5-44 53.5-79 97.4-97.6 22.4-9.7 45.8-14.4 71.3-14.4 24.2-.2 48.3 4.7 70.5 14.4 43.7 18.7 78.3 53.7 96.4 97.6 9.5 22.5 14.1 45.8 14.1 71.4 0 20.5-2.3 39.1-6.8 55.3z"
          />
        </svg>
      </span>
      How can I save my work?
    </button>
  </div>
  <div
    className="h5 u-mb30"
  >
    You’re reviewing
  </div>
  <h1 />
  <p
    className="lead-paragraph"
  >
    Select any dimension to start your review. We recommend starting with content and moving to utility, quality, and efficacy, but you can complete the four dimensions in any order.
  </p>
  <p>
    You can complete all dimensions in one sitting or over the course of many sessions. You’ll be able to print or save  a summary for each dimension as you finish it, and then print or save a final summary for the overall review.
  </p>
  <ul
    className="o-dimension-selection-bar"
  >
    <li
      className="o-dimension-section-bar_dimension"
    >
      <button
        className="o-dimension-section-bar_button Content"
        onClick={[Function]}
      >
        <div
          className="o-dimension-section-bar_selector"
        >
          <span
            className="o-dimension-section-bar_selector-title"
          >
            <span
              className="a-icon a-icon__large a-icon__space-after"
            >
              <svg
                className="cf-icon-svg"
                viewBox="0 0 651.7 1200"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M30 161c-16.5 0-30 13.5-30 30v827.8c0 16.5 13.5 30 30 30h591.7c16.5 0 30-13.5 30-30V343.7L469 161H30zm389.6 60v134.8c0 19.9 16.3 36.2 36.2 36.2h135.9V988.8H60V221h359.6z"
                />
                <path
                  d="M123.8 768.6h394.8v50H123.8zM123.8 644h394.8v50H123.8zM123.8 519.5h394.8v50H123.8z"
                />
                <circle
                  cx="194"
                  cy="382.3"
                  r="60"
                />
              </svg>
            </span>
            Content
             
            <span
              className="o-dimension-section-bar_selector-status"
            />
          </span>
          <span
            className="o-dimension-section-bar_selector-text"
          >
            Covers core knowledge and skills in national standards
          </span>
        </div>
      </button>
      <ul
        className="o-dimension-section-bar_dimension-details"
      >
        <li
          className="h6"
        >
          6 criteria
        </li>
        <li
          className="h6"
        >
          <span
            aria-hidden="true"
          >
            est. time 60 min.
          </span>
          <span
            className="u-visually-hidden"
          >
            estimated time 60 minutes
          </span>
        </li>
      </ul>
    </li>
    <li
      className="o-dimension-section-bar_dimension"
    >
      <button
        className="o-dimension-section-bar_button Utility"
        onClick={[Function]}
      >
        <div
          className="o-dimension-section-bar_selector"
        >
          <span
            className="o-dimension-section-bar_selector-title"
          >
            <span
              className="a-icon a-icon__large a-icon__space-after"
            >
              <svg
                className="cf-icon-svg"
                viewBox="0 0 834 1200"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M804.4 505.7L746 495.6c-3.2-.6-6.5-.6-9.7 0-7.8-30.4-19.8-59.5-35.9-86.5 2.7-1.9 5-4.2 6.9-6.9l34.2-48.5c9.5-13.5 7.8-34.1-3.9-45.7l-51.4-51.4c-11.7-11.7-32.2-13.4-45.7-3.9L592 286.9c-2.7 1.9-5 4.2-6.9 6.9-27-16-56.1-28.1-86.5-35.9.6-3.2.6-6.5 0-9.7l-10.1-58.4c-2.8-16.3-18.6-29.6-35.1-29.6h-72.7c-16.5 0-32.3 13.3-35.1 29.6l-10.1 58.4c-.6 3.2-.6 6.5 0 9.7-30.4 7.8-59.5 19.8-86.5 35.9-1.9-2.7-4.2-5-6.9-6.9l-48.5-34.2c-13.5-9.5-34.1-7.8-45.7 3.9L96.5 308c-11.7 11.7-13.4 32.2-3.9 45.7l34.2 48.5c1.9 2.7 4.2 5 6.9 6.9-16 27-28.1 56.1-35.9 86.5-3.2-.6-6.5-.6-9.7 0l-58.4 10.1C13.3 508.5 0 524.3 0 540.8v72.7c0 16.5 13.3 32.3 29.6 35.1L88 658.8c3.2.6 6.5.6 9.7 0 7.8 30.4 19.8 59.5 35.9 86.5-2.7 1.9-5 4.2-6.9 6.9l-34.2 48.5c-9.5 13.5-7.8 34.1 3.9 45.7l51.4 51.4c11.7 11.7 32.2 13.4 45.7 3.9l48.5-34.2c2.7-1.9 5-4.2 6.9-6.9 27 16 56.1 28.1 86.5 35.9-.6 3.2-.6 6.5 0 9.7l10.1 58.4c2.8 16.3 18.6 29.6 35.1 29.6h72.7c16.5 0 32.3-13.3 35.1-29.6l10.1-58.4c.6-3.2.6-6.5 0-9.7 30.4-7.8 59.5-19.8 86.5-35.9 1.9 2.7 4.2 5 6.9 6.9l48.5 34.2c13.5 9.5 34.1 7.8 45.7-3.9l51.4-51.4c11.7-11.7 13.4-32.2 3.9-45.7l-34.2-48.5c-1.9-2.7-4.2-5-6.9-6.9 16-27 28.1-56.1 35.9-86.5 3.2.6 6.5.6 9.7 0l58.4-10.1c16.3-2.8 29.6-18.6 29.6-35.1v-72.7c.1-16.6-13.2-32.4-29.5-35.2zM417 719.2c-78.4 0-142-63.6-142-142s63.6-142 142-142 142 63.6 142 142-63.6 142-142 142z"
                />
              </svg>
            </span>
            Utility
             
            <span
              className="o-dimension-section-bar_selector-status"
            />
          </span>
          <span
            className="o-dimension-section-bar_selector-text"
          >
            Supports effective teaching
          </span>
        </div>
      </button>
      <ul
        className="o-dimension-section-bar_dimension-details"
      >
        <li
          className="h6"
        >
          5 criteria
        </li>
        <li
          className="h6"
        >
          <span
            aria-hidden="true"
          >
            est. time 60 min.
          </span>
          <span
            className="u-visually-hidden"
          >
            estimated time 60 minutes
          </span>
        </li>
      </ul>
    </li>
    <li
      className="o-dimension-section-bar_dimension"
    >
      <button
        className="o-dimension-section-bar_button Quality"
        onClick={[Function]}
      >
        <div
          className="o-dimension-section-bar_selector"
        >
          <span
            className="o-dimension-section-bar_selector-title"
          >
            <span
              className="a-icon a-icon__large a-icon__space-after"
            >
              <svg
                className="cf-icon-svg"
                viewBox="0 0 969.7 1200"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M843.8 592.5l94.6-87.5c52.5-48.5 37.3-95.1-33.7-103.6l-128-15.2-129-15.3-81.5-176.6-27-58.5c-30-64.9-79-64.9-108.9 0l-27 58.5L322 371l-129.1 15.3-128 15.2c-71 8.4-86.1 55-33.6 103.6l47.3 43.7 47.3 43.7 95.4 88.3L196 808.2l-12.6 63.2-12.6 63.2c-13.9 70.1 25.7 98.9 88.1 64l225.9-126.4 225.9 126.4c62.4 34.9 102 6.1 88.1-64l-12.6-63.2-37.9-190.7 95.5-88.2z"
                />
              </svg>
            </span>
            Quality
             
            <span
              className="o-dimension-section-bar_selector-status"
            />
          </span>
          <span
            className="o-dimension-section-bar_selector-text"
          >
            Presents clear, accurate, and objective information
          </span>
        </div>
      </button>
      <ul
        className="o-dimension-section-bar_dimension-details"
      >
        <li
          className="h6"
        >
          4 criteria
        </li>
        <li
          className="h6"
        >
          <span
            aria-hidden="true"
          >
            est. time 30 min.
          </span>
          <span
            className="u-visually-hidden"
          >
            estimated time 30 minutes
          </span>
        </li>
      </ul>
    </li>
    <li
      className="o-dimension-section-bar_dimension"
    >
      <button
        className="o-dimension-section-bar_button Efficacy"
        onClick={[Function]}
      >
        <div
          className="o-dimension-section-bar_selector"
        >
          <span
            className="o-dimension-section-bar_selector-title"
          >
            <span
              className="a-icon a-icon__large a-icon__space-after"
            >
              <svg
                className="cf-icon-svg"
                viewBox="0 0 651.7 1200"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M651.7 367.7v-24l-91.4-91.4-66.6-66.6L469 161H35c-19.3.1-34.9 15.7-35 35v817.8c.1 19.3 15.7 34.9 35 35h581.7c19.3-.1 34.9-15.7 35-35V367.7zm-60 59v562.1H60V221h359.6v134.8c.1 20 16.2 36.2 36.2 36.2h135.9v34.7z"
                />
                <path
                  d="M279.6 802.8l-77.8-93.4c-14.1-17-39.4-19.3-56.3-5.1-17 14.1-19.3 39.4-5.1 56.3l114.5 137.5c7.6 9.1 18.9 14.4 30.7 14.4 1.1 0 2.3-.1 3.4-.2 13-1.1 24.7-8.5 31.2-19.9l204.7-354.5c11.2-19.1 4.8-43.6-14.3-54.7s-43.6-4.8-54.7 14.3c-.1.2-.2.3-.3.5l-176 304.8z"
                />
              </svg>
            </span>
            Efficacy
             
            <span
              className="o-dimension-section-bar_selector-status"
            />
          </span>
          <span
            className="o-dimension-section-bar_selector-text"
          >
            Has validated impact on knowledge, skills, or behaviors
          </span>
        </div>
      </button>
      <ul
        className="o-dimension-section-bar_dimension-details"
      >
        <li
          className="h6"
        >
          1-3 criteria
        </li>
        <li
          className="h6"
        >
          <span
            aria-hidden="true"
          >
            est. time 30 min.
          </span>
          <span
            className="u-visually-hidden"
          >
            estimated time 30 minutes
          </span>
        </li>
      </ul>
    </li>
  </ul>
  <div />
</div>
`;
